docs/plugins/gatsby-remark-relative-linker/functions.js

```diff
@@ -0,0 +1,59 @@
+const sibling = "./";
+const parent = "../";
+const UNPREFIXED = -1;
+const wasSeen = {};
+
+function* depthCheck(n) {
+  for (let i = n; i > -1; i--) yield i;
+}
+
+function escapeRegExp(string) {
+  //https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping
+  return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+}
+
+function buildDepthCheckRegex(depth) {
+  if (depth === 0) return new RegExp("^" + escapeRegExp(sibling));
+  return new RegExp("^" + escapeRegExp(parent.repeat(depth)));
+}
+
+function doUpdateDepth(linkText, forDepth, isIndex = false) {
+  let noPrefixRegex = /^[\w-_]+/;
+
+  if (linkText.match(noPrefixRegex)) {
+    return updateLinkText(linkText, UNPREFIXED, isIndex);
+  }
+
+  let depthCheckRegex = buildDepthCheckRegex(forDepth);
+  let result = linkText.match(depthCheckRegex);
+
+  if (result) {
+    return updateLinkText(linkText, forDepth, isIndex);
+  }
+}
+
+function updateLinkText(linkText, depth, index) {
+  switch (index) {
+    case true:
+      return depth === UNPREFIXED || 0 ? sibling + linkText : linkText;
+    case false:
+      return depth === UNPREFIXED || 0
+        ? parent + linkText
+        : linkText.replace(
+            buildDepthCheckRegex(depth),
+            parent.repeat(depth + 1)
+          );
```

Comment:
in the case false scenario, would 'parent + linkText' to be good enough (always pre-fix ../) ?

Reply:
Yes, all sibling links on a non-index page need the `../` prefix.